pkg/monitoring/persistent_volumes_collector.go

```diff
@@ -0,0 +1,119 @@
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+)
+
+var (
+	log = logf.Log.WithName("persistent-volume-info-collector")
+
+	metrics = map[MetricsKey]*metric{
+		PersistentVolumeInfo: newMetric(
+			MetricOptsList[PersistentVolumeInfo].Name, MetricOptsList[PersistentVolumeInfo].Help,
+			[]string{"pvc_name", "namespace", "pv_name", "volumemode", "volumeattributes"},
+			prometheusTypeStringToValueType(MetricOptsList[PersistentVolumeInfo].Type),
+		),
+	}
+)
+
+// PersistentVolumeInfoCollector collects information about PersistentVolumes managed by CDI
+type PersistentVolumeInfoCollector struct {
+	client *kubernetes.Clientset
+}
+
+// SetupPersistentVolumesCollector registers PersistentVolumeInfoCollector with Prometheus
+func SetupPersistentVolumesCollector(client *kubernetes.Clientset) PersistentVolumeInfoCollector {
+	c := PersistentVolumeInfoCollector{
+		client: client,
+	}
+	prometheus.MustRegister(c)
+	return c
+}
+
+// Collect implements prometheus.Collector interface
+func (c PersistentVolumeInfoCollector) Collect(ch chan<- prometheus.Metric) {
+	c.collectPersistentVolumeInfo(ch)
+}
+
+// Describe implements prometheus.Collector interface
+func (c PersistentVolumeInfoCollector) Describe(ch chan<- *prometheus.Desc) {
+	for _, metric := range metrics {
+		ch <- metric.desc
+	}
+}
+
+func (c PersistentVolumeInfoCollector) collectPersistentVolumeInfo(ch chan<- prometheus.Metric) {
+	pvcs, err := c.getPVCsManagedByCDI()
+	if err != nil {
+		log.Error(err, "Failed to get PVCs managed by CDI")
+		return
+	}
+
+	for _, pvc := range pvcs {
+		c.createVolumeInfoMetric(ch, pvc)
+	}
+}
+
+func (c PersistentVolumeInfoCollector) getPVCsManagedByCDI() ([]v1.PersistentVolumeClaim, error) {
+	pvcs, err := c.client.CoreV1().PersistentVolumeClaims("").List(
+		context.TODO(),
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", common.AppKubernetesManagedByLabel, "cdi-controller"),
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return pvcs.Items, nil
+}
+
+func (c PersistentVolumeInfoCollector) createVolumeInfoMetric(ch chan<- prometheus.Metric, pvc v1.PersistentVolumeClaim) {
+	pv, err := c.client.CoreV1().PersistentVolumes().Get(context.Background(), pvc.Spec.VolumeName, metav1.GetOptions{})
+	if err != nil {
+		log.Error(err, "Failed to get PV for PVC", "pvc", pvc.Name)
+		return
+	}
+
+	m, err := prometheus.NewConstMetric(
+		metrics[PersistentVolumeInfo].desc,
+		metrics[PersistentVolumeInfo].mType,
+		1,
+		pvc.Name, pvc.Namespace,
+		pvc.Spec.VolumeName,
+		string(*pvc.Spec.VolumeMode),
+		buildVolumeAttributes(pv),
+	)
+
+	if err != nil {
+		log.Error(err, "Failed to create metric", "metric", metrics[PersistentVolumeInfo].desc)
+		return
+	}
+	ch <- m
+}
+
+func buildVolumeAttributes(pv *v1.PersistentVolume) string {
+	volumeAttributes := ""
+
+	if pv.Spec.PersistentVolumeSource.CSI == nil {
+		return volumeAttributes
+	}
+
+	for k, v := range pv.Spec.PersistentVolumeSource.CSI.VolumeAttributes {
```

Comment:
I don't think we want to do this, the volume attributes could contain sensitive information and we don't want to expose them.

Reply:
Well, the rbd:rxbounce attribute its what I'm mostly looking for, so I can look for just some specific attributes